<!-- Create the launch file for ROS -->
<launch>
	<arg name="speed" default="1.0"/>
	<arg name="file_name" default="no_name"/>	
	
	<node name="generatore" pkg="simulator" type="generator.py" output="screen">
		<param name="node_name" value="generatore"/>
		<param name="next_element" value="switch"/>
		<param name="gen_freq" value="2"/>
		<param name="speed" value="$(arg speed)"/>
		<param name="num_messages" value="600"/>
	</node>

	<node name="switch" pkg="simulator" type="switch2.py" output="screen">
		<param name="node_name" value="switch"/>
		<param name="modality" value="split"/>
		<param name="topic1" value="coda1"/>
		<param name="topic2" value="coda2"/>
	</node>

	<node name="coda1" pkg="simulator" type="coda.py" output="screen">
		<param name="node_name" value="coda1"/>
		<param name="next_element" value="coda3"/>
		<param name="num_servers" value="1"/>
		<param name="server_time" value="1.5"/>
		<param name="speed" value="$(arg speed)"/>
	</node>

	<node name="coda2" pkg="simulator" type="coda.py" output="screen">
		<param name="node_name" value="coda2"/>
		<param name="next_element" value="coda3"/>
		<param name="num_servers" value="2"/>
		<param name="server_time" value="1"/>
		<param name="speed" value="$(arg speed)"/>
	</node>

	<node name="coda3" pkg="simulator" type="coda.py" output="screen">
		<param name="node_name" value="coda3"/>
		<param name="next_element" value="end_node"/>
		<param name="num_servers" value="3"/>
		<param name="server_time" value="0.8"/>
		<param name="speed" value="$(arg speed)"/>
	</node>

	<node name="end_node" pkg="simulator" type="end_node.py" output="screen">
		<param name="node_name" value="end_node"/>
		<param name="modality" value="events"/>
		<param name="speed" value="$(arg speed)"/>
		<param name="file_name" value="$(arg file_name)"/>
	</node>

</launch>


